Objects/object.c

```diff
@@ -1715,10 +1715,16 @@ _PyObject_GenericGetAttrWithDict(PyObject *obj, PyObject *name,
             dict = (PyObject *)_PyObject_GetManagedDict(obj);
         }
         else {
+            Py_BEGIN_CRITICAL_SECTION(obj);
+#ifdef DISABLE_GIL
+            PyObject **dictptr = _Py_atomic_load_ptr(_PyObject_ComputedDictPointer(obj));
+#else
             PyObject **dictptr = _PyObject_ComputedDictPointer(obj);
+#endif
             if (dictptr) {
                 dict = *dictptr;
```

Comment:
Here is where the atomic load should be. (And, as Sam said in the issue, we want `_Py_atomic_load_ptr_acquire` instead of plain old `_Py_atomic_load_ptr`.)